app/src/main/java/org/simple/clinic/sync/indicator/SyncIndicatorViewController.kt

```diff
@@ -40,18 +41,39 @@ class SyncIndicatorViewController @Inject constructor(
 
   private fun updateIndicatorView(events: Observable<UiEvent>): Observable<UiChange> {
     val screenCreated = events.ofType<SyncIndicatorViewCreated>()
-    val syncResultStream = lastSyncGroupResult.asObservable()
+    val syncStateStream = lastSyncState
+        .asObservable()
         .distinctUntilChanged()
-        .filterAndUnwrapJust()
-
-    return Observables.combineLatest(screenCreated, syncResultStream) { _, result ->
-      val state = when (result) {
-        SUCCESS -> Synced
-        FAILURE -> SyncPending
-        SYNCING -> Syncing
-      }
-      { ui: Ui -> ui.updateState(state) }
+        .filter { it.lastSyncProgress != null }
+
+    return Observables
+        .combineLatest(screenCreated, syncStateStream)
+        { _, stateStream ->
+          val indicatorState = when (stateStream.lastSyncProgress!!) {
+            SUCCESS -> syncedSuccessState(stateStream.lastSyncSuccessTimestamp)
+            FAILURE -> SyncPending
+            SYNCING -> Syncing
+          }
+          { ui: Ui -> ui.updateState(indicatorState) }
+        }
+  }
+
+  private fun syncedSuccessState(timestamp: Instant?): SyncIndicatorState {
+    if (timestamp == null) {
```

Comment:
Can `timestamp` ever be `null` when this is invoked?

Reply:
I realized this later and made the change in my next PR.
I will do it here itself.